### ISSUE TITLE
"Edit element" option improvements

### ISSUE THREAD

**Issue by FrancRos31** [musescore/MuseScore#25271]
### Issue type

See the title.

### Description with steps to reproduce

Few ideas to improve "Edit element" option when right-clicking something on the score:

(from @MarcSabatella comment)
I'd be very much in favor of seeing some useful indication that you are in edit mode and what it does. Like, a popup that appears showing clicking arrow buttons (in addition to enabling the cursor keys). Or even just some additional highlighting of the element being edited (a colored background, say).

(Editing this report if someone encounter some problem).

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242570931, revision: github-musescore-musescore-3130f97

### Regression

I was unable to check

### Operating system

Windows 10 Pro

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by bkunda** [comment]
I suspect this is a relic of "edit mode". We should probably remove it if it's not functional.

**Comment by MarcSabatella** [comment]
Hmm, I though I had responded but I'm not seeing it.  Anyhow - Edit mode *is* a thing.  It continues to be the way to enable the cursor keys to nudge elements.  So for instance, if you select a note and then press "Right", the selection changes to the next node, but if you first press F2 to enter Edit mode, then when you press "Right", the note moves 0.5 sp to the right.  This is correct behavior, at least until someone designs a system to replace it.

Although see https://github.com/musescore/MuseScore/issues/24721 and https://github.com/musescore/MuseScore/issues/13438 - there is some inconsistency and definite room for improvement in how the nudge distance is determined, even within the current design.

**Comment by FrancRos31** [comment]
> if you first press F2 to enter Edit mode, then when you press "Right", the note moves 0.5 sp to the right.

Actually, pressing F2 doesn't do anything too (F2 still the default shortcut).

https://github.com/user-attachments/assets/1b1d7f63-4640-47a1-b0c3-fef761dd37ba

**Comment by MarcSabatella** [comment]
Your video doesn’t show you trying the cursor keys after entering edit mode - again, that is the whole point, to enable the cursor keys. Merely entering edit mode doesn’t do anything in itself.

**Comment by FrancRos31** [comment]
You're right, my bad for not reading the handbook. Thanks @MarcSabatella, I'm gonna close this, study the handbook and return here if I find something worth reporting.

**Comment by MarcSabatella** [comment]
FWIW, I'd be very much in favor of seeing some useful indication that you are in edit mode and what it does.  Like, a popup that appears showing clicking arrow buttons (in addition to enabling the cursor keys).  Or even just some additional highlighting of the element being edited (a colored background, say).

**Comment by bkunda** [comment]
Unless I'm missing something obvious, I don't think we should improve edit mode at all. I think we should get rid of it. 

It's completely undiscoverable, poorly named, and buries a really helpful interaction: i.e. nudging.

I think nudging of elements should simply be allowed with a modifier key (e.g. Alt/Option, or similar) plus the arrow keys. Without having to enter a specific mode. 

Having to enter a mode to do this adds an unnecessary layer of friction, and isn't something you encounter in most other graphics editing applications. 

It's also completely unnecessary to tell the user they are in "edit mode" when they edit text elements.  

This issue is more of a feature request than anything. I'd be curious to learn more about what is being requested here:

> Or even just some additional highlighting of the element being edited (a colored background, say).

If this means a clear indication of what's happening when you are nudging something, then I'd say that we can certainly solve that without edit mode.

**Comment by MarcSabatella** [comment]
That’s fine too of course.  Although I’m not sure that having modifier-combo-plus-cursor to nudge is more discoverable than having and edit mode option that appears in the context menu - really it’s less so.  So that would be something consider in an alternate design.  But getting rid of the extra step of having to enter (and exit) a special edit mode is a win for sure, and of course the naming made no sense to most people anyhow.

**Comment by FrancRos31** [comment]
> I'd be curious to learn more about what is being requested here

My bad, too much confusion in this report. 
So, I'll try to explain: I select a note, right click, press "Edit element", nothing. I try with other elements, only with text frames something happens. So, I report this issue saying that "Edit element" mode doesn't do anything for elements that aren't text frames. Then, @MarcSabatella commented that Edit element exists and works, so I try again and I see that it actually works. So, I think "maybe this mode needs some improvements" and I edit the issue.  
My intention was to make you aware that "Edit element" has problems, not to add confusion. 
I have no problem with whatever solution you'll choose.

**Comment by stevage** [comment]
I also came to report the UX bug that "Edit element" does nothing. It's a big surprise to me to discover that it does *something* in a few limited contexts.

What I expected it to do: switch the sidebar to the "Properties" tab (opening it if necessary) and focus it. (This would be a good way to make the Properties tab a bit more discoverable). It is the most obvious, intuitive way for a user to try to edit the properties of something: right click.
 
It is definitely incorrect UX bug to have a context menu item that does nothing. If there's no "edit mode" for notes, rests etc, then at the very least the context menu item should not be there.

Perhaps to combine these two ideas:

- remove the "Edit element" menu item in contexts where it does nothing
- add a "Properties" menu item in contexts where it can do something (similar to "Bar properties..." which opens a pop-up), as described above.

**Comment by MarcSabatella** [comment]
There certainly is an edit mode for notes and many other other elements - it's for allowing the cursor cursors to make manual adjustments to the position of the element, as explained above.  It's certainly possible that someday a design will be proposed that allows for keyboard adjustments without requiring a separate mode.  But for now at least, it serves a very important purpose.

**Comment by stevage** [comment]
Hopefully we can agree that the current behaviour, which for the average user is indistinguishable from "does nothing", could be greatly improved. There is no feedback of any kind to indicate a state change, and "edit element" is a very vague term considering all the other ways the element can be edited that don't require entering that state.

**Comment by MarcSabatella** [comment]
Certainly.  If Edit mode is kept, there should be visible indication of this, such as adding a handle and/or showing "edit mode" (or whatever other name is chosen) in the status bar.
